net.go

```diff
@@ -69,11 +69,7 @@ func serverAccept(listener net.Listener, wg *sync.WaitGroup, stopper chan bool,
 
 		tlsConn, ok := conn.(*tls.Conn)
 		if !ok {
-			openCounter.Dec(1)
-			errorCounter.Inc(1)
-			logger.Printf("received non-TLS connection from %s? ignoring", conn.RemoteAddr())
-			conn.Close()
-			continue
+			panic("got non-TLS socket from a TLS listener")
```

Comment:
It's an impossible error. If we have a TLS listener, it should always return a TLS connection. If we get a non-TLS connection object, something must be seriously wrong.